### PR TITLE
Update of IPP integration

### PR DESCRIPTION
**Pull request by maver1**
Please merge with: https://github.com/opencv/opencv_3rdparty/pull/34

### This pullrequest changes

- Supported IPP version updated to IPP 2019
- Usability improvements of IPP detection in build routine + wording cleanup
- Removed IPP support of Haar

```
force_builders=docs,linux,windows,macosx
```

### CODE REVIEW

**Comment by lupustr3** [comment]
+1 Thanks, Andrey.

**Comment by alalek** [comment]
IPP detection is broken.
[Missing](http://pullrequest.opencv.org/buildbot/builders/precommit_linux64/builds/18747) lines like these:
```
--   Other third-party libraries:
--     Intel IPP:                   2017.0.3 [2017.0.3]
--            at:                   /build/precommit_linux64/build/3rdparty/ippicv/ippicv_lnx
--     Intel IPP IW:                sources (2017.0.3)
--               at:                /build/precommit_linux64/build/3rdparty/ippicv/ippiw_lnx
```

**Comment by maver1** [comment]
@alalek, thank you. Fixed.

File: modules/core/src/channels.cpp [hunk 1]
```diff
@@ -341,8 +341,8 @@ namespace cv
 {
 static bool ipp_extractChannel(const Mat &src, Mat &dst, int channel)
 {
-#ifdef HAVE_IPP_IW
-    CV_INSTRUMENT_REGION_IPP();
+#ifdef HAVE_IPP_IW_LL
+    CV_INSTRUMENT_REGION_IPP()
```
Comment (alalek) [inline modules/core/src/channels.cpp]: Why is ';' ending removed?

MSVS editor doesn't like that.

Comment (maver1) [reply modules/core/src/channels.cpp]: Perhaps it was done unintentionally, returned semicolon back.

**Pull request merged**